### PR TITLE
Rework logging colors

### PR DESCRIPTION
**Pull request by KristjanESPERANTO**
- Replacing old package `colors` by drop-in replacement `ansis`
- Rework `console-stamp` config to show all Log outputs in same color (errors = red, warnings = yellow, debug = blue background (only for the label), info = blue)
- This also fixes `npm run config:check` (broken since 6097547c103be1bc89c572e5bd2be50b342967fa)

Feel free to let me know if the PR is too big and you want me to do individual PRs for the changes.

Before:
before

After:
after

### CODE REVIEW

File: js/utils.js [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
+ * By KristjanESPERANTO https://github.com/KristjanESPERANTO
  * MIT Licensed.
  */
 const execSync = require("node:child_process").execSync;
-const colors = require("colors/safe");
 const Log = require("logger");
 const si = require("systeminformation");
 
 module.exports = {
-	colors: {
```
Comment (rejas) [inline js/utils.js:13]: would it be possible for a module developer to use Utils.color in his module? Just wondering, dont have time to test...

**Review by rejas — COMMENTED** [review]
just two small comments (always in favor of updating old stuff :-)

File: js/utils.js [hunk 2]
```diff
@@ -1,21 +1,14 @@
 /* MagicMirror²
  * Utils
  *
- * By Rodrigo Ramírez Norambuena https://rodrigoramirez.com
+ * By KristjanESPERANTO https://github.com/KristjanESPERANTO
```
[the remark this answers is not in the document]
Comment (KristjanESPERANTO) [reply js/utils.js:4]: Good point. I would also like something like `by The MagicMirror Community` or something similar. If that's clean from a licensing point of view.

File: js/utils.js (hunk 1 above)
Comment (KristjanESPERANTO) [reply js/utils.js:13]: Yeah, I think so. That would be an issue for those modules. I'll check tomorrow if any modules on the official list does that.

**Comment by KristjanESPERANTO** [comment]
Just added screenshots :slightly_smiling_face:

**Comment by sdetweil** [comment]
as we collect these logs by redirection 95% of the time, these ansi colors are not helpful.

show the pm2 logs output
or the output from npm start >somefile.txt 2>&1

**Comment by KristjanESPERANTO** [comment]
I set the PR to draft to check the use of Utils.colors in the 3rd party modules and to check why the tests fail.

**Comment by KristjanESPERANTO** [comment]
> these ansi colors are not helpful

In some cases, I debug directly with the help of the console. That's why I think the colors make sense. They sholdn't do any harm either. And in general, my changes are just a more consistent implementation of the previous approach.

> show the pm2 logs output
> or the output from npm start >somefile.txt 2>&1

Before:
```bash
$ npm start >somefile.txt 2>&1
[14.01.2024 23:17.53.726] [LOG]   This is Log.log.
[14.01.2024 23:17.53.726] [WARN]  This is Log.warn.
[14.01.2024 23:17.53.727] [ERROR] This is Log.error.
[14.01.2024 23:17.53.727] [INFO]  This is Log.info.
[14.01.2024 23:17.53.727] [DEBUG] This is Log.debug.
```

After:
```bash
$ npm start >somefile.txt 2>&11
[2024-01-14 23:16:10.903] [LOG]   This is Log.log. 
[2024-01-14 23:16:10.904] [WARN]  This is Log.warn. 
[2024-01-14 23:16:10.905] [ERROR] This is Log.error. 
[2024-01-14 23:16:10.905] [INFO]  This is Log.info. 
[2024-01-14 23:16:10.905] [DEBUG] This is Log.debug. 
```

**Comment by sdetweil** [comment]
typically there are  ansi escape sequences in the output

File: js/utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (MichMich) [reply js/utils.js:4]: We can remove those author headers. It's a thing of the past. 👍🏻

File: js/utils.js (hunk 1 above)
Comment (KristjanESPERANTO) [reply js/utils.js:13]: Okay, I first searched all 1116 modules to see if they import `utils`. This is not the case :slightly_smiling_face:

Then I searched all modules to see if they use `colors`. I found one and it doesn't have it listed in its dependencies. For this one module, the change would therefore be relevant. So I have just created [a PR for the module's repository to fix this](https://github.com/djey47/MMM-IDF-LOCALTRANSPORT/pull/17).

Even if the PR for the module is not accepted, I would say that this is not a blocker for this PR here.

File: js/utils.js [hunk 3]
```diff
@@ -32,6 +24,9 @@ module.exports = {
 			systemDataString += `\n### VERSIONS: electron: ${process.versions.electron}; used node: ${staticData["versions"]["node"]}; installed node: ${installedNodeVersion}; npm: ${staticData["versions"]["npm"]}; pm2: ${staticData["versions"]["pm2"]}`;
 			systemDataString += `\n### OTHER:    timeZone: ${Intl.DateTimeFormat().resolvedOptions().timeZone}; ELECTRON_ENABLE_GPU: ${process.env.ELECTRON_ENABLE_GPU}`;
 			Log.info(systemDataString);
+
+			// Return is currently only for jest
+			return systemDataString;
```
Comment (KristjanESPERANTO) [inline js/utils.js:29]: @khassel I only added this return for the jest test (in `utils_spec.js`). It works, but it doesn't feel quite right. What do you think?

File: js/utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (KristjanESPERANTO) [reply js/utils.js]: I have now removed the line in this one file. Do we want a PR that removes the author line everywhere?

[the remark this answers is not in the document]
Comment (rejas) [reply js/utils.js]: A seperate PR would be nice

[the remark this answers is not in the document]
Comment (khassel) [reply js/utils.js]: see https://github.com/MagicMirrorOrg/MagicMirror/issues/3358 for further discussions ...

**Review by rejas — APPROVED** [review]

**Pull request merged**